opencog/atoms/base/Atom.h

```diff
@@ -173,6 +173,11 @@ class Atom
 
     virtual ~Atom();
 
+    virtual const std::string& getName() const {
+        static std::string empty_string("");
+        return empty_string;
```

Comment:
yes, I noticed that after adding the comment.  However, silently failing when there is a coding error seems like poor design, which is why I suggest having a throw, instead.   There is also a very minor performance boost, because `try..catch` is faster than `if (empty==...)` for those cases where the programmer wants to explicitly check for the error condition.